contracts/cw-core/src/contract.rs

```diff
@@ -312,35 +321,47 @@ pub fn execute_update_proposal_modules(
     env: Env,
     sender: Addr,
     to_add: Vec<ModuleInstantiateInfo>,
-    to_remove: Vec<String>,
+    to_disable: Vec<String>,
 ) -> Result<Response, ContractError> {
     if env.contract.address != sender {
         return Err(ContractError::Unauthorized {});
     }
 
-    for addr in to_remove {
+    let disable_count = to_disable.len() as u32;
+    for addr in to_disable {
         let addr = deps.api.addr_validate(&addr)?;
-        PROPOSAL_MODULES.remove(deps.storage, addr);
+        let mut module = PROPOSAL_MODULES
+            .load(deps.storage, addr.clone())
+            .map_err(|_| ContractError::ProposalModuleDoesNotExist {
+                address: addr.clone(),
+            })?;
+
+        if module.status == ProposalModuleStatus::Disabled {
+            return Err(ContractError::ModuleAlreadyDisabled {
+                address: module.address,
+            });
+        }
+
+        module.status = ProposalModuleStatus::Disabled {};
+        PROPOSAL_MODULES.save(deps.storage, addr, &module)?;
     }
 
+    // If disabling this module will cause there to be no active modules, return error.
+    // We don't check the active count before disabling because there may erroneously be
+    // modules in to_disable which are already disabled.
+    ACTIVE_PROPOSAL_MODULE_COUNT.update(deps.storage, |count| {
+        if count <= disable_count && to_add.is_empty() {
+            return Err(ContractError::NoActiveProposalModules {});
+        }
+        Ok(count - disable_count)
```

Comment:
Really love that you set this to `count - disable_count` and not `count - disable_count + to_add.len()`. Tiny detail, but means the state is correct during the dispatch of proposal module instantiation messages. Good stuff.